consensus/src/metrics_safety_rules.rs

```diff
@@ -31,9 +31,14 @@ impl MetricsSafetyRules {
         let sr_waypoint = consensus_state.waypoint();
         let proofs = self
             .storage
-            .retrieve_epoch_change_proof(sr_waypoint.version())
-            .expect("Unable to retrieve Waypoint state from Storage");
-        self.initialize(&proofs)
+            .retrieve_epoch_change_proof(sr_waypoint.version());
+        if let Err(e) = proofs {
+            panic!(
+                "Unable to retrieve Waypoint state from storage, encountered Error:{}",
+                e
+            )
+        }
+        self.initialize(&proofs.unwrap())
```

Comment:
For this one, I think just returning an InternalError(format!(....)) would suffice. I don't think this is particularly more readable than expect. what do we gain by this indirection?